metals/src/main/scala/scala/meta/internal/metals/BuildTargets.scala

```diff
@@ -379,17 +255,20 @@ final class BuildTargets(
         // else it can be a source file inside a jar
         val fromJar = jarPath(source)
           .flatMap { jar =>
-            allBuildTargetIds.find { id =>
+            allBuildTargetIdsInternal.find { case (_, id) =>
               targetJarClasspath(id).exists(_.contains(jar))
             }
           }
-        fromJar.foreach(addSourceItem(source, _))
-        fromJar
+        for ((data0, id) <- fromJar)
+          data0.addSourceItem(source, id)
+        fromJar.map(_._2)
```

Comment:
```suggestion
        fromJar.map { case (data, id) =>
          data.addSourceItem(source, id)
          id
        }
```
Unless you don't want to mix these, we could just iterate through them just once. Take it or leave it.

Reply:
Sure. I just saw it's just an `Option`, so it shouldn't change much things.